src/components/disclosure/disclosure.module.css

```diff
@@ -17,6 +17,7 @@
   cursor: pointer;
   display: flex;
   padding: 16px;
+  position: relative;
```

Comment:
in case you're interested, I have come to this conclusion after a few experiments/back&forth with the focus ring:
- the parent should have a `isolation: isolate` declararation (it's used to create a new stacking context, needed to have the pseudo element below text/icon but not the parent container)
- the pseudo-element should have a `z-index: -1` so is below the content (in case some day you may need to add a background to the pseudo-element)